src/herder/TransactionQueue.cpp

```diff
@@ -702,25 +702,53 @@ TransactionQueue::rebroadcast()
         else
         {
             auto& tx = it->first->mTx;
-            if (opsToFlood < tx->getNumOperations())
+            bool broadcasted =
+                it->first->mBroadcasted;
+
+            if (broadcasted)
```

Comment:
As we do incremental broadcasting, we want to get to the interesting transactions quickly.

The work needed by floodgate to "no-op" is quite substantial: it needs to compute a cryptographic hash of the message (which we could store in the TransactionQueue instead of booleans -> doesn't really simplify anything) and perform lookups of that hash in the floodmap (that, we cannot avoid).